components/DragResizeRotate.vue

```diff
@@ -0,0 +1,220 @@
+            });
+        }
+    }
+
+    onMounted(() => {
+        updateBB();
+        counterRotation.value = `${-angle.value}deg`;
+        comicStore.bus.on('updateText', (obj) => {
+            if (obj.id == props.eId) {
+                text.value = obj.text;
+                fontSize.value = obj.fontSize;
+            }
+        });
+    });
+
+    function updateCornersPosition() {
+        tL = { x: self.value.left, y: self.value.top };
+        tR = { x: self.value.left + self.value.width, y: self.value.top };
+        bR = { x: self.value.left + self.value.width, y: self.value.top + self.value.height };
+        bL = { x: self.value.left, y: self.value.top + self.value.height };
+    }
+
+    // d=√((x2 – x1)² + (y2 – y1)²).
+    function updateBB() {
+        updateCornersPosition();
+        let d1 = Math.sqrt(Math.pow(tL.x - bR.x, 2) + Math.pow(tL.y - bR.y, 2));
+        let d2 = Math.sqrt(Math.pow(tR.x - bL.x, 2) + Math.pow(tR.y - bL.y, 2));
+
+        maxDiagonal.value = `${Math.max(d1, d2)}px`;
+    }
+</script>
+
+<template>
+    <VueDragResizeRotate
+        :z="z"
+        :w="w"
+        :h="h"
+        :eId="eId"
+        class-name-active="element--active"
+        ref="self"
+        :disableUserSelect="true"
+        :x="pos.currPos().x"
+        :y="pos.currPos().y"
+        :parent="true"
+        :rotatable="true"
+        :resizable="true"
+        :draggable="true"
+        :r="angle"
+        @rotating="rotating"
+        @resizing="isResizing = true"
+        @activated="() => (elementActive = true)"
+        @deactivated="() => (elementActive = false)"
+        @dragstop="updatePosition(eId)"
+        @resizestop="resize(eId)"
+        @rotatestop="updateRotation(eId)"
+    >
+        <div class="edition-menu--container">
+            <EditionMenu
+                v-if="elementActive && !isRotating && !isResizing"
+                @mirror-horizontal-event="updateMirroring(eId, (direction = 'x'))"
+                @mirror-vertical-event="updateMirroring(eId, (direction = 'y'))"
+                @delete-event="$emit('deleteEvent', eId)"
+            />
+        </div>
+        <div
+            tabindex="-1"
+            class="text"
+            :class="{ mirror: mirroredHorizontal || mirroredVertical }"
+            v-if="fontSize != 0"
+            @dblclick="comicStore.setCurrentElement(props.element)"
+        >
+            <p class="text__content" :style="{ fontSize: fontSize + 'px' }">
+                {{ text }}
+            </p>
+        </div>
+
+        <img
+            :src="url"
+            :alt="altText"
+            :class="{ mirror: mirroredHorizontal || mirroredVertical }"
+            v-if="fontSize == 0"
+        />
+    </VueDragResizeRotate>
+</template>
+
+<style lang="scss" scoped>
+    img {
+        width: 100%;
+        height: 100%;
+    }
+
+    .element--active {
+        border: $border-width solid $info;
+    }
+
+    .edition-menu--container {
```

Comment:
Regarding the convention, this would be just `edition-menu`, since you don't have any parent where you would be a child. -- would also be the syntax for a modifier.

Reply:
that's changed!